docs(checkbox): update checkbox documentation (#1894)
IgniteUI/igniteui-docfx: en/components/checkbox.md

    <iframe id="checkbox-sample-2-iframe" data-src='{environment:demosBaseUrl}/data-entries/checkbox-sample-2' width="100%" height="100%" seamless="" frameBorder="0" class="lazyload"></iframe>
</div>
<div>
    <button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="checkbox-sample-2-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
</div>

#### Label Positioning

You can position the label using the checkbox's [`labelPosition`]({environment:angularApiUrl}/classes/igxcheckboxcomponent.html#labelposition) property:

```html
<igx-checkbox labelPosition="before"></igx-checkbox>
```

If the `labelPosition` is not set, the label will be positioned after the checkbox.

#### Indeterminate Checkbox

In addition to the checked and unchecked states, there is a third state a checkbox can be in: **indeterminate**. In this state the checkbox is neither checked, nor unchecked. This is set using the checkbox's [`indeterminate`]({environment:angularApiUrl}/classes/igxcheckboxcomponent.html#indeterminate) property:

```html
<igx-checkbox [indeterminate]="true"></igx-checkbox>
```

> We can create an app that has a list of tasks that need to be done and one master checkbox that's going to be checked only if all the tasks beloware completed. We're going to upgrade the previous sample. Starting with the template:

Reviewer: ```suggestion
We can create an app that has a list of tasks that need to be done and one master checkbox that's going to be checked only if all the tasks are completed. Let's update the previous sample. Starting with the template:
```

```html
<!-- app.component.html -->

> <igx-checkbox [readonly]="true" [(ngModel)]="masterCheckbox.checked" [(indeterminate)]="masterCheckbox.indeterminate" (click)="toggleAll()">

Reviewer: ```suggestion
<igx-checkbox 
    [readonly]="true" 
    [(ngModel)]="masterCheckbox.checked" 
    [(indeterminate)]="masterCheckbox.indeterminate" 
    (click)="toggleAll()"
>
```

All done
</igx-checkbox>
<igx-checkbox class="tasks" *ngFor="let task of tasks" [(ngModel)]="task.done">
    {{ task.description }}
</igx-checkbox>
```

> Next, we're going to indent the subtasks, so it's more visual that they're dependable on each other.

Reviewer: ```suggestion
Next, we're going to indent the subtasks, so it's more visual that they are part of the same group.
```
